File: PaperScope/frontend/src/api/papers.ts
export interface Paper {
    id: number;
    title: string;
    authors: string;
    source: string;
    published_date: string;
    url: string;
    abstract: string | null;
    ai_summary: string | null;
    matched_keywords: string | null;
  }
  
  const API_BASE_URL = process.env.NEXT_PUBLIC_API_URL || "http://localhost:8000/api";
  
  export async function fetchPapers(skip: number = 0, limit: number = 50): Promise<Paper[]> {
    try {
      const response = await fetch(`${API_BASE_URL}/papers/?skip=${skip}&limit=${limit}`, {
        // Fetch fresh data explicitly or rely on Next.js default cache
        cache: 'no-store', 
      });
      if (!response.ok) {
        throw new Error('Failed to fetch papers');
      }
      return response.json();
    } catch (error) {
      console.error("Error fetching papers:", error);
      return [];
    }
  }

File: PaperScope/backend/api/papers.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from typing import List, Optional

from backend.database import get_db
from backend.models.paper import Paper
from pydantic import BaseModel, ConfigDict
from datetime import datetime

router = APIRouter(prefix="/papers", tags=["papers"])

# Pydantic Schemas for response validation
class PaperResponse(BaseModel):
    id: int
    title: str
    authors: str
    source: str
    published_date: datetime
    url: str
    abstract: Optional[str]
    ai_summary: Optional[str]
    matched_keywords: Optional[str]

    model_config = ConfigDict(from_attributes=True)

@router.get("/", response_model=List[PaperResponse])
def get_papers(skip: int = 0, limit: int = 50, db: Session = Depends(get_db)):
    """Fetch the latest crawled papers from the database."""
    papers = db.query(Paper).order_by(Paper.published_date.desc()).offset(skip).limit(limit).all()
    return papers
